### PR TITLE
ExpressRoutePort cmdlets

### PR DESCRIPTION
**Pull request by srividyathirumala**
## Description

Adding cmdlets for ExpressRoutePort resource

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRoutePortTests.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+using System;
+using Microsoft.Azure.ServiceManagemenet.Common.Models;
+using Microsoft.WindowsAzure.Commands.ScenarioTest;
+using Microsoft.WindowsAzure.Commands.Test.Utilities.Common;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Commands.Network.Test.ScenarioTests
+{
+    public class ExpressRoutePortTestsGenerated : RMTestBase
```
Comment (EvgenyAgafonchikov) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRoutePortTests.cs]: Please drop *Generated suffix. Test framework uses class name to find corresponding ps1 file.

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRoutePortsLocationTests.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+using Microsoft.WindowsAzure.Commands.Test.Utilities.Common;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Commands.Network.Test.ScenarioTests
+{    
+    public class ExpressRoutePortsLocationTestsGenerated : RMTestBase
+    {
+        public XunitTracingInterceptor _logger;
+        public ExpressRoutePortsLocationTestsGenerated(Xunit.Abstractions.ITestOutputHelper output)
```
Comment (EvgenyAgafonchikov) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRoutePortsLocationTests.cs]: Please drop *Generated suffix. Test framework uses class name to find corresponding ps1 file.

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRoutePortTests.ps1 [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+Test creating new ExpressRoutePort
+#>
+function Test-ExpressRoutePortCRUD
+{
+    # Setup
+    $rgname = Get-ResourceGroupName
+    $rglocation = Get-ProviderLocation ResourceManagement
+    $rname = Get-ResourceName
+    $location = Get-ProviderLocation "Microsoft.Network/ExpressRoutePorts"
+	$location = "centraluseuap"
```
Comment (EvgenyAgafonchikov) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRoutePortTests.ps1]: hardcoded location.
If you'd like to set preferred location, please use 3rd parameter of another function: `Get-Location "Microsoft.Network" "ExpressRouteCircuits" "centraluseuap"`

**Comment by number213** [comment]
> Please run tests in recording mode and add recordings to project as "Copy if Newer" or "Copy Always"

"Copy if Newer" is preferable, "Copy Always" would overwrite all new recordings in `bin` each time you rebuild the project

File: tools/StaticAnalysis/Exceptions/AzureRM.Network/SignatureIssues.csv [hunk 4]
```diff
@@ -270,3 +270,11 @@
...
+"c:\wkpc\PsPri-OnDemand\src\Package\Release\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll","Microsoft.Azure.Commands.Network.NewAzureRmExpressRoutePort","New-AzureRmExpressRoutePort","1","8410","Parameter BandwidthInGbps of cmdlet New-AzureRmExpressRoutePort does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
+"c:\wkpc\PsPri-OnDemand\src\Package\Release\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll","Microsoft.Azure.Commands.Network.RemoveAzureRmExpressRoutePort","Remove-AzureRmExpressRoutePort","1","8600","Cmdlet 'Remove-AzureRmExpressRoutePort' has no defined output type.","Add an OutputType attribute that declares the type of the object(s) returned by this cmdlet. If this cmdlet returns no output, please set the output type to 'bool' and make sure to implement the 'PassThru' parameter."
```
Comment (EvgenyAgafonchikov) [inline tools/StaticAnalysis/Exceptions/AzureRM.Network/SignatureIssues.csv]: Remove related entries (this one and one more below) are unnecessary as Remove has return type and PassThru param in your code

File: tools/StaticAnalysis/Exceptions/AzureRM.Network/SignatureIssues.csv [hunk 5]
```diff
@@ -270,3 +270,11 @@
...
+"c:\wkpc\PsPri-OnDemand\src\Package\Release\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll","Microsoft.Azure.Commands.Network.NewAzureExpressRouteCircuitCommand","New-AzureRmExpressRouteCircuit","1","8410","Parameter BandwidthInGbps of cmdlet New-AzureRmExpressRouteCircuit does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
+"c:\wkpc\PsPri-OnDemand\src\Package\Release\ResourceManager\AzureResourceManager\AzureRM.Network\Microsoft.Azure.Commands.Network.dll","Microsoft.Azure.Commands.Network.NewAzureExpressRouteCircuitCommand","New-AzureRmExpressRouteCircuit","1","8510","Cmdlet 'New-AzureRmExpressRouteCircuit' has multiple parameter sets, but no defined default parameter set.","Define a default parameter set in the cmdlet attribute."
```
Comment (EvgenyAgafonchikov) [inline tools/StaticAnalysis/Exceptions/AzureRM.Network/SignatureIssues.csv]: It would be better to add default parameter set code to avoid cmdlet usage issues than add these entries to exclusions

**Comment by praries880** [comment]
@srividyathirumala Can you link the design review for the cmdlets in the PR?

File: src/ResourceManager/Network/Commands.Network/ExpressRouteCircuit/NewAzureExpressRouteCircuitCommand.cs [hunk 6]
```diff
@@ -70,20 +70,35 @@ public class NewAzureExpressRouteCircuitCommand : ExpressRouteCircuitBaseCmdlet
         public string SkuFamily { get; set; }
 
         [Parameter(
-            Mandatory = true,
+            ParameterSetName = "ServiceProvider",
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ExpressRouteCircuit/NewAzureExpressRouteCircuitCommand.cs]: since you are introducing poaram sets, you need to declare the default param set for the cmdlet.

File: src/ResourceManager/Network/Commands.Network/AzureRM.Network.psd1 [hunk 7]
```diff
@@ -389,7 +389,13 @@ CmdletsToExport = 'Add-AzureRmApplicationGatewayAuthenticationCertificate',
                'Set-AzureRmNetworkInterfaceTapConfig', 
                'Remove-AzureRmNetworkInterfaceTapConfig', 
                'Get-AzureRmVirtualNetworkTap', 'New-AzureRmVirtualNetworkTap', 
-               'Remove-AzureRmVirtualNetworkTap', 'Set-AzureRmVirtualNetworkTap'
+               'Remove-AzureRmVirtualNetworkTap', 'Set-AzureRmVirtualNetworkTap',
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/AzureRM.Network.psd1]: you need to add the `az` versions of the cmdlets to the [`Az.Network.psd1`](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network/Az.Network.psd1) file as well

File: src/ResourceManager/Network/Commands.Network.Test/packages.config [hunk 8]
```diff
@@ -9,19 +9,21 @@
   <package id="Microsoft.Azure.Management.Authorization" version="2.0.0" targetFramework="net452" />
   <package id="Microsoft.Azure.Management.Compute" version="22.0.0" targetFramework="net452" />
   <package id="Microsoft.Azure.Management.ContainerInstance" version="1.1.0" targetFramework="net452" />
-  <package id="Microsoft.Azure.Management.Network" version="19.6.0" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.Network" version="19.5.1-preview" targetFramework="net452" />
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network.Test/packages.config]: Is this SDK a locally generated SDK version? If So.. is it generated using the code checked into the [October branch](https://github.com/Azure/azure-rest-api-specs/tree/network-october-release) for NRP swagger changes?

It also need sto include the changes in the SDK its replacing (namely 19.6.0)

Comment (praries880) [reply src/ResourceManager/Network/Commands.Network.Test/packages.config]: @srividyathirumala 
The latest guidance to generate the network SDKs is to generate on based off of : 
https://github.com/Azure/azure-rest-api-specs/tree/network-october-release
If you have swagger changes in your private repo kindly merge them into the October branch and then generate the SDK off of it. 

@MikhailTryakhov can give you more details if the need be.

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj [hunk 9]
```diff
@@ -43,8 +43,8 @@
     <RunCodeAnalysis>false</RunCodeAnalysis>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="Microsoft.Azure.Management.Network, Version=19.6.0.0">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.19.6.0\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
+    <Reference Include="Microsoft.Azure.Management.Network, Version=19.6.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: can you revert this line to :
``` suggestion
<Reference Include="Microsoft.Azure.Management.Network, Version=19.6.1.0>
```

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj [hunk 10]
```diff
@@ -65,6 +65,15 @@
     <Reference Include="Microsoft.Azure.Management.ContainerInstance">
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.ContainerInstance.1.1.0\lib\net452\Microsoft.Azure.Management.ContainerInstance.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Azure.Test.HttpRecorder, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: revert these changes... the recorder n client runtime come from common targets...

File: src/ResourceManager/Network/Commands.Network.Test/packages.config [hunk 11]
```diff
@@ -9,19 +9,21 @@
...
   <package id="Microsoft.Azure.Management.Storage" version="6.1.0-preview" targetFramework="net45" />
   <package id="Microsoft.Azure.Test.Framework" version="3.0.0" targetFramework="net452" />
-  <package id="Microsoft.Azure.Test.HttpRecorder" version="1.6.8" targetFramework="net45" />
+  <package id="Microsoft.Azure.Test.HttpRecorder" version="1.13.0" targetFramework="net452" />
   <package id="Microsoft.Bcl" version="1.1.9" targetFramework="net45" />
   <package id="Microsoft.Bcl.Async" version="1.0.168" targetFramework="net45" />
   <package id="Microsoft.Bcl.Build" version="1.0.14" targetFramework="net45" />
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.3" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.28" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.14" targetFramework="net452" />
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network.Test/packages.config]: revert this and the next line

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj [hunk 12]
```diff
@@ -58,9 +58,19 @@
       <HintPath>..\..\..\packages\Microsoft.Azure.KeyVault.Core.1.0.0\lib\net40\Microsoft.Azure.KeyVault.Core.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.Network, Version=19.6.0.0">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.19.6.0\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
+    <Reference Include="Microsoft.Azure.Management.Network, Version=19.6.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: change this back to :
``` suggestion
<Reference Include="Microsoft.Azure.Management.Network, Version=19.6.1.0>
```

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj [hunk 13]
```diff
@@ -58,9 +58,19 @@
...
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.19.6.1-preview\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: revert the following changes

**Comment by praries880** [comment]
@srividyathirumala  kindly resolve the merge conflicts

**Comment by maddieclayton** [comment]
@praries880 Can we merge this PR into the October branch? Or do we want to wait until the branch is merged back to preview?
